krod/core/vector_store.py

```
@@ -87,87 +87,135 @@ def add_document(self, text: str, metadata: Dict[str, Any] = None) -> str:
+                    })
+                    
+                next_page_offset = scroll_result[1]
+                if next_page_offset is None:
+                    break
+                    
+            return results[:limit]
         except Exception as e:
-            self.logger.error(f"Failed to add documents: {str(e)}")
-            raise RuntimeError(f"Failed to add documents: {str(e)}")
+            self.logger.error(f"Metadata search failed: {str(e)}")
+            raise RuntimeError(f"Metadata search failed: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)